cpp/src/arrow/util/cpu_info.cc

```diff
@@ -200,44 +93,59 @@ bool RetrieveCacheSize(int64_t* cache_sizes) {
           GetModuleHandle("kernel32"), "GetLogicalProcessorInformation");
 
   if (!func_pointer) {
-    return false;
+    return;
```

Comment:
Can we log warnings in case of unexpected errors in this function?

Reply:
Done